pydicom/pixel_data_handlers/numpy_handler.py

```diff
@@ -112,11 +112,15 @@ def get_pixeldata(dicom_dataset):
         expected_length *= dicom_dataset.SamplesPerPixel
     if dicom_dataset.BitsAllocated > 8:
         expected_length *= (dicom_dataset.BitsAllocated // 8)
-    if length_of_pixel_array != expected_length:
+    padded_length = expected_length
+    if expected_length & 1:
+        padded_length += 1
+    if length_of_pixel_array != padded_length:
         raise AttributeError(
             "Amount of pixel data %d does not "
             "match the expected data %d" %
-            (length_of_pixel_array, expected_length))
+            (length_of_pixel_array, padded_length))
+    pixel_array = pixel_array[:expected_length]
     if should_change_PhotometricInterpretation_to_RGB(dicom_dataset):
```

Comment:
I would do this only if expected_length != padded_length, as it copies the array unnecessary otherwise.

Reply:
I am pretty sure numpy slices don't copy, but there is no harm in being explicit...

Reply:
I wasn't sure so I looked it up - but I may have looked at the wrong place. So if you are sure, just leave it...